source/core/index.ts

```diff
@@ -731,10 +731,11 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 
 			const updatedOptions = new Options(undefined, undefined, this.options);
 
-			const shouldBeGet = statusCode === 303 && updatedOptions.method !== 'GET' && updatedOptions.method !== 'HEAD';
-			if (shouldBeGet || updatedOptions.methodRewriting) {
-				// Server responded with "see other", indicating that the resource exists at another location,
-				// and the client should request it from that location via GET or HEAD.
+			if ((updatedOptions.methodRewriting && ![307, 308].includes(statusCode))
+				|| (statusCode === 303 && !['GET', 'HEAD'].includes(updatedOptions.method))) {
```

Comment:
Do not nest `if`s like this. Instead, define a constant.

Reply:
```
const shouldBeGet = statusCode === 303 && updatedOptions.method !== 'GET' && updatedOptions.method !== 'HEAD';
const canRewrite = statusCode !== 307 && statusCode !== 308;
const shouldRewrite = updatedOptions.methodRewriting && canRewrite;
if (shouldBeGet || shouldRewrite) {
```

Reply:
done, with improved variable names